devito/ir/equations/algorithms.py

```diff
@@ -141,6 +141,10 @@ def lower_exprs(expressions, **kwargs):
             indices = [(lower_exprs(a) + o) for a, o in
                        zip(i.indices, f._size_nodomain.left)]
 
+            # Substitute spacing (spacing only used in own dimension)
+            indices = [i.xreplace({d.spacing: 1, -d.spacing: -1})
+                       for i, d in zip(indices, f.dimensions)]
```

Comment:
why is this here now?

Reply:
I asked you about it some weeks ago in #development on Slack. We agreed that "we would have talked about it when I would have opened this PR".

IIRC, the reason is that the new optional GPU pass, `buffering`, introduces Arrays to implement the buffers. These buffers, just like any other Functions, require lowering -- they have a halo, may be using SubDimensions, and so on. 

So, I construct these copy-back `Eq`s , from the buffer to the original Function , e.g.

```
 Eq(u[time + dt, x, y], ub[sb0 + dt, x, y])]
```

`ub` is the buffer -- here accessed via an indexed Array, whose `dt` also needs to be turned into 1, just like any `dt` (any spacing symbol) appearing in any Indexed expressed with the Indexed notation

More generally, if we ever wanted to expose Array to the user API, or simply use it in the higher layers of the compiler (just like I'm doing here), I need them to be lowered too.

Does this make sense?

Reply:
EDIT: and yeah, the buffering tests in `test_lower_exprs.py` start failing without it

Reply:
But the `lower_exprs` the line just above should have indexified and do the replacement already so this really shouldn't be necessary

Reply:
no, the timing/ordering is different (you can easily see it if you drop the line and run any of the buffering tests in test_lower_exprs.py)